api/errors/error.go

```diff
@@ -0,0 +1,25 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker engine but are not tied to any specific top-level component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+const errGroup = "engine"
+
+var (
+	// ErrorCodeGeneric is generated when there is no better error message
+	// available. This will typically happen when some low-level function
+	// is called and we just need to bubble the unknown error back up to
+	// the API. We should try to avoid using this.
+	ErrorCodeGeneric = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
Then let's make it more specific. I really don't like have a bunch of the same things defined because you slightly disagree. We have an error code defined globally for this exact use case. Use it.